### ISSUE TITLE
Generate potoken and visitordata from sig helper?

### ISSUE THREAD

**Issue by unixfox** [iv-org/inv_sig_helper#10]
@techmetx11 someone found on how to generate poToken without a browser: https://github.com/YunzheZJU/youtube-po-token-generator

Do you think it would be somewhat possible to integrate this into sig helper project?

More resources: https://gist.github.com/LuanRT/5792d7e9b1d74210362530058151762e

Cc @SamantazFox

**Comment by techmetx11** [comment]
The program works by emulating the DOM API, intercepting JavaScript functions, and making a fake `ytcfg` object for it all to work.
I'm going to figure out how to implement this within the program itself

**Comment by unixfox** [comment]
What about using https://napi.rs for using jsdom npm library?

**Comment by unixfox** [comment]
Yeah. I have looked too bad there is no way to use the npm library jsdom directly in rust.

There is a lookalike library in rust but it's probably incomplete: https://github.com/spider-rs/jsdom

**Comment by techmetx11** [comment]
I've decided to just make inv_sig_helper run a extractor script written in Node.JS, and parse and return (maybe cache aswell) the result back to the users
This means it'll be the slowest operation in the whole server, so I hope it'll only be ran occassionally or so

**Comment by unixfox** [comment]
Won't this make building for cross platform more difficult? How are you going to achieve this?

I was wondering if we could instead run the jsdom library using node from rust: https://austinpoor.com/blog/js-in-rs

**Comment by techmetx11** [comment]
> Won't this make building for cross platform more difficult? How are you going to achieve this?
> 
> I was wondering if we could instead run the jsdom library using node from rust: https://austinpoor.com/blog/js-in-rs

Nevermind, I'll figure out how to use Deno's runtime, although that means we'll have 2 JS interpreters inside the program

**Comment by unixfox** [comment]
@techmetx11 can you give us your matrix username? We would like to invite you to a private room for more in depth info about this implementation

**Comment by techmetx11** [comment]
> @techmetx11 can you give us your matrix username? We would like to invite you to a private room for more in depth info about this implementation

@techmetx11:digitalprivacy.diy
